55 cutadapt

dhoogest opened the pull request.

Commit:
add cutadapt current version, bump barcodecop in docker requirements

Commit:
add cutadapt step, params to its test params

Commit:
add cutadapt tag prefix to build instructions

Commit:
remove cutadapt tag from gh actions

Commit:
add --allow-empty argument to barcodecop steps

Commit:
publish cutadapt simple tsv output and detailed .json

nhoffman committed:
Merge branch 'master' into 55-cutadapt

nhoffman committed:
pin dependencies in requirements.txt

- Docker image builds from requirements.txt at top level

Commit:
update default params for full adaptor seqs

Commit:
move requirements back to /docker for now, install xsv

Commit:
extract cutadaput output counts into a stacked file with sampleid

- TODO: figure out how to aggregate counts files into a channel for
input to `ljoin` script

nhoffman commented on main.nf:
```diff
@@ -562,6 +605,7 @@ if(params.containsKey("bidirectional") && params.bidirectional){
 
 process join_counts {
     input:
+        file("raw.csv") from raw_counts_concat
```
Are cutadapt counts making it into here? i don't see a reference to ``cutadapt_counts_concat``, which I assume would hold the input file.

dhoogest replied:
No not yet - I haven't figured out how to pack/unpack a channel of files ... have some nextflow learnin' to do.  Alternatively could just define `cutadapt_counts_concat` as empty in the null condition.

dhoogest replied:
Okay not ideal, but after running out of patience trying to figure out how to concat then expand a non-deterministic list of counts_concat channels I went with the approach seen [here](https://github.com/nhoffman/dada2-nf/pull/58/commits/51dd38bc82fb66184c51440e33a25b11b56d848e). Basically defines an empty `cutadapt_counts_concat` channel and then tests channel content to set the join command. Happy to have this adjusted if either @nhoffman or @crosenth knows how to better achieve.

dhoogest replied:
Oops - please hold, think the logic is backward.

Commit:
add cutadapt_counts_concat to join_counts process

- conditionally compose command for now, since I couldn't figure out
an elegant way to do this with channel concatenation/expansion

Commit:
turns out .empty() wasn't doing what I thought it was

Commit:
ugly conditional strategy just to get it to work

Commit:
remove some un-needed comments/publishing

dhoogest commented:
@nhoffman okay I think this branch is functionally complete now, although could be likely improved a bit in terms of how counts files are packed/unpacked. I have not added specific tests to inspect/verify the adaptor trimming, but would note that the `params-its.json` is run as part of the automated testing, and therefore both the 'with cutadapt_params' and 'without' conditions are currently included.

Commit:
consolidate nextflow.config to single ghcr registry def

nhoffman commented on main.nf:
```diff
@@ -118,6 +118,43 @@ process plot_quality {
+        file("cutadapt_counts.csv") into cutadapt_counts_concat
+
+    """
+    stack_cutadapt_counts.sh *.cutadapt.tsv | sed -e 's/out_reads/cutadapt/' | xsv select -d '\t' sampleid,cutadapt  > cutadapt_counts.csv
```
Is there a reason not to move all of the necessary processing into ``stack_cutadapt_counts.sh`` and write directly to an output file?

dhoogest replied:
No -  I'm weak enough in bash that I didn't want to try to do the column selection without the help of xsv is the main reason I wrote it this way. I guess the command above could happen in the bash script directly (moving `sed` and `xsv` into the bash script), but not sure if this preferable to leaving commands in the main.nf.

nhoffman replied:
Ok, not important to make any changes now. May revisit later.

nhoffman committed:
Merge pull request #61 from nhoffman/fix-singularity-config

Fix singularity container in nextflow.config
